pandas/core/frame.py

```diff
@@ -6445,96 +6445,33 @@ def melt(
     # ----------------------------------------------------------------------
     # Time series-related
 
+    @doc(
+        Series.diff,
+        klass="Dataframe",
+        extra_params="axis : {0 or 'index', 1 or 'columns'}, default 0\n    "
+        "Take difference over rows (0) or columns (1).",
+        other_klass="Series",
+        examples="Difference with previous row\n\n>>> df = pd.DataFrame({'a'"
+        ": [1, 2, 3, 4, 5, 6],\n...                    'b': [1, 1, 2, 3, 5, "
+        "8],\n...                    'c': [1, 4, 9, 16, 25, 36]})\n>>> df\n "
+        "  a  b   c\n0  1  1   1\n1  2  1   4\n2  3  2   9\n3  4  3  16\n4  "
+        "5  5  25\n5  6  8  36\n\n>>> df.diff()\n     a    b     c\n0  NaN  "
+        "NaN   NaN\n1  1.0  0.0   3.0\n2  1.0  1.0   5.0\n3 \n1.0  1.0   7.0"
+        "\n4  1.0  2.0   9.0\n5  1.0  3.0  11.0\n\nDifference with previous "
+        "column\n\n>>> df.diff(axis=1)\n    a    b     c\n0 NaN  0.0   0.0\n"
+        "1 NaN -1.0   3.0\n2 NaN -1.0   7.0\n3 NaN -1.0  13.0\n4 NaN  0.0  2"
+        "0.0\n5 NaN  2.0  28.0\n\nDifference with 3rd previous row\n\n>>> df"
+        ".diff(periods=3)\n     a    b     c\n0  NaN  NaN   NaN\n1  NaN  NaN"
+        "   NaN\n2  NaN  NaN   NaN\n3 \n3.0  2.0  15.0\n4  3.0  4.0  21.0\n5"
+        "  3.0  6.0  27.0\n\nDifference with following row\n\n>>> df.diff(pe"
+        "riods=-1)\n     a    b     c\n0 -1.0  0.0  -3.0\n1 -1.0 -1.0  -5.0"
+        "\n2 -1.0 -1.0  -7.0\n3 -1.0 -2.0  -9.0\n4 -1.0 -3.0 -11.0\n5  NaN  "
+        "NaN   NaN\n\nOverflow for input dtype\n\n>>> df = pd.DataFrame({'a'"
+        ": [1, 0]}, dtype=np.uint8)\n>>> df.diff()\n       a\n0    NaN\n1  2"
+        "55.0",
```

Comment:
that's unreadable, use `"""` and make the `\n` actual line breaks, so this can be read.